addons/visiblefavor/visiblefavor.lua

```diff
@@ -0,0 +1,127 @@
+
+defaults = {}
+defaults.enabled = true
+defaults.display_mode = 'favor'
+defaults.effect_mode = 'debuff'
+
+settings = config.load(defaults)
+
+function check_incoming_chunk(id, original, modified, injected, blocked)
+    if id == 0x037 then
+        refresh = true
+    elseif settings.enabled and id == 0x00E then
+        if refresh and settings.display_mode ~= 'all' then
+            local player = windower.ffxi.get_player()
+            pet_index = windower.ffxi.get_mob_by_index(player.index).pet_index
+            if settings.display_mode == 'favor' then
+                for _,buff_id in pairs(player.buffs) do
+                    if buff_id == 431 then
+                        favor_buff = true
+                        break
+                    end
+                    favor_buff = false
+                end
+            end
+        end
+        local npc_index = original:unpack('H', 9)
+        if settings.display_mode == 'all' or settings.display_mode == 'self' and pet_index == npc_index or settings.display_mode == 'favor' and favor_buff and pet_index == npc_index then
+            local npc_model = original:unpack('H', 0x33)
+            if model_to_effect[npc_model] then
+                local effect = model_to_effect[npc_model]
+                if settings.effect_mode == 'debuff' then
+                    effect = effect + 8
+                end
+                return modified:sub(1, 38) .. string.char(effect) .. modified:sub(40)
+            end
+        end
+    end
+end
+
+function visiblefavor_command(...)
+    local arg = {...}
+    if #arg == 1 and arg[1]:lower() == 'toggle' then
+        settings.enabled = not settings.enabled
+    elseif #arg == 1 and arg[1]:lower() == 'on' then
+        settings.enabled = true
+    elseif #arg == 1 and arg[1]:lower() == 'off' then
+        settings.enabled = false
+    elseif #arg == 2 and arg[1]:lower() == 'display' and arg[2]:lower() == 'all' then
+        settings.display_mode = 'all'
+    elseif #arg == 2 and arg[1]:lower() == 'display' and arg[2]:lower() == 'self' then
+        settings.display_mode = 'self'
+    elseif #arg == 2 and arg[1]:lower() == 'display' and arg[2]:lower() == 'favor' then
+        settings.display_mode = 'favor'
+    elseif #arg == 2 and arg[1]:lower() == 'effect' and arg[2]:lower() == 'buff' then
+        settings.effect_mode = 'buff'
+    elseif #arg == 2 and arg[1]:lower() == 'effect' and arg[2]:lower() == 'debuff' then
+        settings.effect_mode = 'debuff'
+    else
+        windower.add_to_chat(167, 'Command usage:')
+        windower.add_to_chat(167, '    vf toggle/on/off')
+        windower.add_to_chat(167, '    vf display all/self/favor')
+        windower.add_to_chat(167, '    vf effect buff/debuff')
+        return
+    end
+    windower.add_to_chat(207, 'visiblefavor: enable = %s, display = \31\200%s\30\1, effect = \31\200%s\30\1':format(settings.enabled and '\31\204yes\30\1' or '\31\167no\30\1', settings.display_mode, settings.effect_mode))
```

Comment:
You mean in the string? Yeah that drives me nuts, but I thought that was the only way to represent non-printable characters in a string in Lua?